twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,322 @@
+ angular.module('cloudberry.cache', ['leaflet-directive', 'cloudberry.common' ])
+ .service('Cache', function( $window, $http, $compile){
+
+  var cachedCityPolygonTree = rbush();
+  var cachedRegion ;
+  var cacheSize = 0;
+  var insertedTreeIDs = new Set();
+  var cacheThreshold = 7500;//hard limit
+  var DeleteTarget = 0;
+  var DeletedCount = 0;
+  var preFetchDistance = 25;
+
+
+  /*Call will happen from map controller and this functions sees whether a required data is present in cache or not
+  if not gets from midleware*/
```

Comment:
Let's use the following:

Map controller calls this function and this function checks whether a requested region is present in the cache or not. If not, it gets the requested region data from the middleware.